src/app/auth/login/login.component.ts

```diff
@@ -0,0 +1,40 @@
+import {Component, OnInit} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from "@angular/forms";
+import {AuthService} from "../../shared/services/auth.service";
+import {IUserInfo} from "../../shared/interfaces/user-info.interface";
+import {Router} from "@angular/router";
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+  loginForm: FormGroup;
+  errorMsg?: boolean;
+
+  constructor(private fb: FormBuilder, private authService: AuthService, private router: Router) {
+    this.loginForm = this.fb.group({
+      username: ['', Validators.required],
+      password: ['', Validators.required],
+    })
+  }
+
+  ngOnInit(): void {
+  }
+
+  login() {
+    let loginData: IUserInfo = {
+      username: this.loginForm.value.username,
+      password: this.loginForm.value.password
+    }
+    this.authService.login(loginData).subscribe(
+      data => {
+        localStorage.setItem('user token', data.access_token);
+        this.router.navigate(['home']);
+      }, err => {
+        this.errorMsg = true;
+      },
+    )
```

Comment:
if you noticed you will find this error showing 

> Deprecated symbol used, consult docs for better alternative 

that's because if you want to listen to error, you should use object listener like that
```suggestion
    this.authService.login(loginData).subscribe({
      next: data => {
        localStorage.setItem('user token', data.access_token);
        this.router.navigate(['home']);
      },
      error: err => {
        this.errorMsg = true;
      },
    });
```
please do the same in register